usaspending_api/disaster/v2/views/object_class/spending.py

```diff
@@ -0,0 +1,139 @@
+                | Q(gross_outlay_amount_by_program_object_class_cpe__lt=0)
+            ),
+            Q(disaster_emergency_fund__in=self.def_codes),
+            Q(object_class__isnull=False),
+            self.all_closed_defc_submissions,
+        ]
+
+        annotations = {
+            **self.universal_annotations(),
+            "obligation": Coalesce(
+                Sum(
+                    Case(
+                        When(
+                            self.final_period_submission_query_filters,
+                            then=F("obligations_incurred_by_program_object_class_cpe"),
+                        ),
+                        default=Value(0),
+                    )
+                ),
+                0,
+            ),
+            "outlay": Coalesce(
+                Sum(
+                    Case(
+                        When(
+                            self.final_period_submission_query_filters,
+                            then=F("gross_outlay_amount_by_program_object_class_cpe"),
+                        ),
+                        default=Value(0),
+                    )
+                ),
+                0,
+            ),
+        }
+
+        # Assuming it is more performant to fetch all rows once rather than
+        #  run a count query and fetch only a page's worth of results
+        return (
+            FinancialAccountsByProgramActivityObjectClass.objects.filter(*filters)
+            .values("object_class__major_object_class", "object_class__major_object_class_name",)
+            .annotate(**annotations)
+            .values(*annotations.keys())
+        )
+
+    @property
+    def award_queryset(self):
+        filters = [
+            Q(disaster_emergency_fund__in=self.def_codes),
+            Q(object_class__isnull=False),
+            self.all_closed_defc_submissions,
+        ]
+
+        annotations = {
+            **self.universal_annotations(),
+            "obligation": Coalesce(Sum("transaction_obligated_amount"), 0),
+            "outlay": Coalesce(
+                Sum(
+                    Case(
+                        When(self.final_period_submission_query_filters, then=F("gross_outlay_amount_by_award_cpe")),
+                        default=Value(0),
+                    )
+                ),
+                0,
+            ),
+        }
+
+        # Assuming it is more performant to fetch all rows once rather than
+        #  run a count query and fetch only a page's worth of results
+        return (
+            FinancialAccountsByAwards.objects.filter(*filters)
+            .values("object_class__major_object_class", "object_class__major_object_class_name")
+            .annotate(**annotations)
+            .values(*annotations.keys())
+        )
+
+    def universal_annotations(self):
+        return {
+            "major_code": F("object_class__major_object_class"),
+            "count": Count("object_class_id", distinct=True),
```

Comment:
This count will be skewed because Object Class "uniqueness includes the `direct_reimbursable` flag which isn't a concern for this endpoint. the object classes should be using unique `object_class` values